deployments/qat_plugin/base/daemonset-dpdk/kustomization.yaml

```diff
@@ -0,0 +1,2 @@
+resources:
+  - intel-qat-plugin.yaml
```

Comment:
I'd prefer we don't consider `dpdk` and `kernel` as equal alternatives. `dpdk` should be the default (e.g., `kubectl apply -k deployments/qat_plugin` OR  `deployments/qat_plugin/base` and `kernel` would be an opt-in).

Reply:
@grahamwhaley, @mythi, thanks for the feedback!

Planned actions:
- Rename directories under `overlays` to `dpdk`, `dpdk-debug`, `kernel`, `kernel-debug`.
- Make `kubectl apply -k deployments/qat_plugin` work.

Note that there currently are [security restrictions](https://github.com/kubernetes-sigs/kustomize/issues/1736) in kustomize that
1. block sharing `patches` between kustomizations (even through symbolic links)
2. block loading `resources` that are not in the same directory or below to `kustomize.yaml`. (Again, even symbolic links are blocked.)
3. anyway allow loading `bases` without this restriction, providing a workaround to load resources blocked by (2). Therefore loading `intel-qat-kernel-plugin.yaml` from a "non-secure" directory needs extra `kustomization.yaml` in that directory. @mythi, this is the only reason for existence of `base/daemonset-kernel/kustomization.yaml`, for instance.

Furthermore, there's a (pre)cautious cycle detection that prevents having a parent directory in `bases` even though this would not actually cause a cycle.

It's quite likely that above restrictions disallow having `intel-qat-plugin.yaml` and/or `intel-qat-plugin-config.yaml` directly in `qat_plugin/`or `qat_plugin/base` so that they could be still shared with kernel mode kustomization. I'll see if I can find better option for file locations to emphasize the proriority of dpdk over kernel... but at least `kubectl apply -k deployments/qat_plugin` will work, that's for sure.

Reply:
> Rename directories under overlays to dpdk, dpdk-debug, kernel, kernel-debug.

My suggestion was that we drop kernel and kernel-debug overlays (`intel-qat-kernel-plugin` does not use/need `ConfigMap`).